nixos/modules/services/x11/desktop-managers/xfce4-13.nix

```diff
@@ -0,0 +1,136 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.xserver.desktopManager.xfce4-13;
+in
+
+{
+  options = {
+    services.xserver.desktopManager.xfce4-13 = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enable the Xfce desktop environment.";
+      };
+
+#     thunarPlugins = mkOption {
+#       default = [];
+#       type = types.listOf types.package;
+#       example = literalExample "[ pkgs.xfce4-13.thunar-archive-plugin ]";
+#       description = ''
+#         A list of plugin that should be installed with Thunar.
+#       '';
+#     };
+
+      noDesktop = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Don't install XFCE desktop components (xfdesktop, panel and notification daemon).";
+      };
+
+      extraSessionCommands = mkOption {
+        default = "";
+        type = types.lines;
+        description = ''
+          Shell commands executed just before XFCE is started.
+        '';
+      };
+
+      enableXfwm = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Enable the XFWM (default) window manager.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = with pkgs.xfce4-13 // pkgs; [
+      # Get GTK+ themes and gtk-update-icon-cache
+      gtk2.out
+
+      # Supplies some abstract icons such as:
+      # utilities-terminal, accessories-text-editor
+      gnome3.adwaita-icon-theme
+
+      hicolor-icon-theme
+      tango-icon-theme
+      xfce4-icon-theme
+
+      desktop-file-utils
+      shared-mime-info
+
+      # Needed by Xfce's xinitrc script
+      # TODO: replace with command -v
+      which
+
+      exo
+      garcon
+      gtk-xfce-engine
+      gvfs
+      libxfce4ui
+      tumbler
+      xfconf
+
+      mousepad
+      ristretto
+      xfce4-appfinder
+      xfce4-screenshooter
+      xfce4-session
+      xfce4-settings
+      xfce4-terminal
+
+      thunar # (thunar.override {  thunarPlugins = cfg.thunarPlugins; })
+    # thunar-volman # TODO: drop
+    ] ++ (if config.hardware.pulseaudio.enable
+          then [ xfce4-pulseaudio-plugin xfce4-volumed-pulse ]
+          else [ xfce4-mixer ])
+      # TODO: NetworkManager doesn't belong here
+      ++ optionals config.networking.networkmanager.enable [ networkmanagerapplet ]
+      ++ optionals config.powerManagement.enable [ xfce4-power-manager ]
+      ++ optionals cfg.enableXfwm [ xfwm4 ]
+      ++ optionals (!cfg.noDesktop) [
+        xfce4-panel
+        xfce4-notifyd
+        xfdesktop
+      ];
+
+    environment.pathsToLink = [
+      "/share/xfce4"
+      "/share/themes"
+      "/share/mime"
+      "/share/desktop-directories"
+      "/share/gtksourceview-3.0"
+    ];
+
+    environment.variables = {
+      GDK_PIXBUF_MODULE_FILE = "${pkgs.librsvg.out}/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache";
```

Comment:
IIRC correctly we shouldn't need to set this in the modules anymore.
However that may have been because of wrappers.

Reply:
This file is mostly copy-paste from ```4.12``` made without understanding of every line.
So I have nothing to answer to all 4 of your questions :(
Feel free to improve, this branch is writable

Reply:
We already have `services.xserver.gdk-pixbuf.modulePackages = [ pkgs.librsvg ];` for that.

Reply:
We still need it globally, though, because not everything is wrapped and some icons in CSD are in SVG.